cmake/CMakeLists.txt

```diff
@@ -183,6 +183,7 @@ option(onnxruntime_WEBASSEMBLY_RUN_TESTS_IN_BROWSER "Enable this option to run t
 option(onnxruntime_ENABLE_WEBASSEMBLY_DEBUG_INFO "Enable this option to turn on DWARF format debug info" OFF)
 option(onnxruntime_ENABLE_WEBASSEMBLY_PROFILING "Enable this option to turn on WebAssembly profiling and preserve function names" OFF)
 option(onnxruntime_ENABLE_WEBASSEMBLY_OUTPUT_OPTIMIZED_MODEL "Enable this option to allow WebAssembly to output optimized model" OFF)
+option(onnxruntime_ENABLE_WEBASSEMBLY_FLOAT16 "Enable this option to allow WebAssembly to use fp16" OFF)
```

Comment:
I think we don't need this build option. 

The benefit of this option is that when we have 2 set of artifacts - one built with fp16 one without - it is possible for the non-fp16 version to fallback to a slow CPU implementation so that a fp16 model can get correct results.

This is not really helpful because no one would like to run fp16 on CPU in a production environment, and also we don't want to expend the artifact table to support a new dimension.

So the solution is simple - we don't use this option and we don't build 2 versions of ORT. we can simply register the operators but if the environment does not support f16 we can do some logic from JS side - not in C++ with build time determined flags.